docs/DeveloperGuide.md

```diff
@@ -2,12 +2,361 @@
+<br>
+<img src="./Diagrams/Recurrence/recurrenceSequenceDiagram.png" alt="recurrenceSequenceDiagram" width="800"/>
+<br>
+For the reference frame of 'load from storage', refer to [Storage component](#storage-component). <br>
+For the reference frame of 'add recurring entry', refer to 
+[checkIncomeRecurrence / checkSpendingRecurrence](#checkincomerecurrence--checkspendingrecurrence-method) method. <br>
+
+#### How the Recurrence Component works:<br>
++ Upon running the application by the user, `Storage` component will load the `IncomeList` and `SpendingList` members of
+`Wiagi` to retrieve past data.
++ Both list are then iterated through. Each member of the list is parsed through `Parser` which returns the type of 
+recurrence it is (e.g. `DailyRecurrence`, `null`) which is encapsulated as a `Recurrence` object.
++ If `Recurrence` is not `null` (i.e. a recurring entry), it checks the entry and adds to the `SpendingList` and 
+`IncomeList` if needed. <br>
+
+#### Implementation:
+#### Recurrence class
+The `Recurrence` class is an abstract class that provides the interface for checking `Income` and `Spending` and adding 
+recurring entries into the list. <br>
+The following are the abstract methods defined: <br>
++ `checkSpendingRecurrence`
++ `checkIncomeRecurrence`
+
+The following are child classes of `Recurrence`:
++ `DailyRecurrence`: Handles entries labelled as daily recurring events
++ `MonthlyRecurrence`: Handles entries labelled as monthly recurring events
++ `YearlyRecurrence`: Handles entries labelled as yearly recurring events
+
+##### RecurrenceFrequency enumeration
+The `RecurrenceFrequency` enumeration is used to determine the type of recurring entry of `EntryType` and its child
+classes <br>
+Enumeration constants:
++ `NONE`: Represents a none recurring entry
++ `DAILY`: Represents a daily recurring entry
++ `MONTHLY`: Represents a monthly recurring entry 
++ `YEARLY`: Represents a yearly recurring entry
+
+##### parseRecurrence method
+Class: `Parser` <br>
+Method Signature: <br>
+```
+public static Recurrence parseRecurrence(Type entry)
+```
+Functionality: <br>
+1. Takes in child class of `Type` (i.e. `Spending`, `Income`)
+2. Matches the `reccurenceFrequency` attribute with switch case to determine which `Recurrence` child to return
+3. Returns `DaillyRecurrence`, `MonthlyRecurrence`, `YearlyRecurrence` or `null`(If not a recurring entry).
+
+##### checkIncomeRecurrence / checkSpendingRecurrence method
+Class: `DailyRecurrence`, `MonthlyRecurrence`, `YearlyRecurrence` <br>
+Method Signature: <br>
+```
+@Override
+public void checkIncomeRecurrence(Income recurringIncome, IncomeList incomes)
+@Override
+public void checkSpendingRecurrence(Spending recurringSpending, SpendingList spendings)
+```
+Below illustrates the functionality of the checkIncomeRecurrence method through a sequence diagram <br>
+<br>
+<img src="./Diagrams/Recurrence/addRecurrenceEntry.png" alt="addRecurrenceEntry" width="800"/> <br>
```

Comment:
activation bars should begin with a method call. here, the activation bar under incomes:IncomeList has no method call. This problem occurs in some other sequence diagrams as well
![Screenshot 2024-11-02 233421](https://github.com/user-attachments/assets/06b2d1cb-b35c-432b-8395-845498713d62)